test/new_tests/test_query_apply.py

```diff
@@ -508,3 +509,57 @@ def _no_items_have_been_applied(self):
             assert bins["name"] != "aerospike"
         else:
             assert bins["name"] == "no_set_name"
+
+    def test_fail_if_operations_exist_before_apply(self):
+        """
+        Invoke query.apply() with a stream udf.
+        arguments contain a serialized set.
+        """
+        ops = [
+            operations.increment("testBinName", 1)
+        ]
+
+        try:
+            query_results = (
+                self.as_connection.query(
+                    "test",
+                    "demo",
+                )
+                .add_ops(ops)
+                .apply(
+                    "query_apply_parameters",
+                    "query_params",
+                    [["age", 5], pickle.dumps({"lary", "quinton", "julie", "mark"})],
+                )
+                .results()
+            )
+            assert "Param error should be thrown" == "This code should not be executed"
+        except e.ParamError:
+            pass
```

Comment:
1. You should split up this one line into multiple lines, and use `with pytest.raises(e.ParamError)` to ensure that the error happens. If you use `try` and `except`, the error might not be thrown and the test would still pass, which is not correct

```suggestion
      query = self.as_connection.query("test", "demo")
      query.add_ops(ops)
      query.apply(
          "query_apply_parameters",
          "query_params",
          [["age", 5], pickle.dumps({"lary", "quinton", "julie", "mark"})],
      )
      with pytest.raises(e.ParamError) as excinfo:
          query.results()
      assert excinfo.value.msg == "No operations can be used with query.results"
```

Reply:
I included the wrong test cases in this branch, I removed them and added the correct ones, so these comments should not be relevant anymore.